src/system-module.adoc

```diff
@@ -0,0 +1,257 @@
+[[sec:sysmod]]
+== The System Module
+
+include::styles.adoc[]
+
+The symbols and macros of the system module `$ion` is available everywhere within an Ion document,
+with the version of that module being determined by the spec-version of each segment.
+
+The specific system symbols are largely uninteresting to users since they don't affect the text
+content that they interact with.
+The system macros are more visible, especially to authors of macros.
+
+This chapter catalogs the system-provided macros.
+The examples below use unqualified names, which works assuming no other module exports the same
+name, but the unambiguous form `:$ion:__macro-name__` is always correct.
+
+
+IMPORTANT: This list is not complete. We expect it to grow and evolve as we gain experience
+writing macros.
+
+
+=== Primitive Operators
+
+This section describes operators that cannot be defined as macros.
+
+
+==== Stream Constructors
+
+===== `void`
+
+[{nrm}]
+----
+(void) \-> any?
+----
+
+Produces an empty stream.
+The most common use of this operator is to supply “no value” to a voidable parameter.
+To make such use more readable, the special-case E-expression `(:)` is synonymous to `(:void)`.
+
+
+===== `values`
+
+[{nrm}]
+----
+(values (v any\...)) \-> any*
+----
+
+Produces a stream from any number of arguments, concatenating the streams produced by the nested
+expressions.
+Used to aggregate multiple values or sub-streams to pass to a single argument, or to return
+multiple results. Generally only useful with more than one subexpression.
+
+
+==== Value Constructors
+
+===== `make_string`
+
+[{nrm}]
+----
+(make_string (content text\...)) \-> string
+----
+
+Produces a non-null, unannotated string containing the concatenated content produced by the
+arguments. Nulls and annotations are discarded.
+
+TODO Probably useful to allow some other Ion scalars (at least) to allow type conversion.
+I think this would be most useful for ints, since the binary representation is more compact than
+as characters. Lobs wouldn’t work well, though.
```

Comment:
> I think this would be most useful for ints, since the binary representation is more compact than
as characters.

I'm not sure I followed this. `(make_string 500)` would be going in the wrong direction, wouldn't it?

Reply:
Let's see... in binary the macro variant takes 2 bytes for the system-macro invocation, 3 bytes for the integer.  Versus 4(?) bytes for the 3-character string.  So yeah, not a win for small numbers, but larger numbers should have an increasing advantage, no?

That said, what I had in mind was more along the lines of inserting a variable number into a fixed string, along the lines of the traditional ASIN to URL macro.

Reply:
> Let's see... in binary the macro variant takes 2 bytes for the system-macro invocation, 3 bytes for the integer. Versus 4(?) bytes for the 3-character string. So yeah, not a win for small numbers, but larger numbers should have an increasing advantage, no?

I see, so anywhere you would have written `"7000000"` you could instead invoke a macro that was equivalent to `(make_string 7000000)` to save space. Yep, that checks out.

> That said, what I had in mind was more along the lines of inserting a variable number into a fixed string, along the lines of the traditional ASIN to URL macro.

This is a compelling use case!